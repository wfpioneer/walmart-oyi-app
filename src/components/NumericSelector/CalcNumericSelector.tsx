import React from 'react';
import { Text, View } from 'react-native';
import MaterialCommunityIcon from 'react-native-vector-icons/MaterialCommunityIcons';
import { TouchableOpacity } from 'react-native-gesture-handler';
import styles from './NumericSelector.style';
import COLOR from '../../themes/Color';
import IconButton, { IconButtonType } from '../buttons/IconButton';

interface CalcNumericSelectorProps {
  // eslint-disable-next-line react/require-default-props
  testID?: string;
  isValid: boolean;
  onDecreaseQty(): void;
  onIncreaseQty(): void;
  minValue: number;
  maxValue: number;
  value: number;
  onEndEditing?: () => void;
  onInputPress: () => void;
}

const renderPlusMinusBtn = (name: 'plus' | 'minus', isDisabled: boolean) => (
  <MaterialCommunityIcon
    name={name}
    color={isDisabled ? COLOR.DISABLED_BLUE : COLOR.MAIN_THEME_COLOR}
    size={18}
  />
);

const CalcNumericSelector = (props: CalcNumericSelectorProps): JSX.Element => {
  const {
    testID,
    isValid,
    onDecreaseQty,
    onIncreaseQty,
    minValue,
    maxValue,
    value,
    onEndEditing,
    onInputPress
  } = props;
  const isMinimum = value <= minValue;
  const isMaximum = value >= maxValue;
  return (
    <View
      testID={testID}
      style={[
        styles.updateContainer,
        isValid ? styles.updateContainerValid : styles.updateContainerInvalid
      ]}
    >
      <IconButton
        testID="decreaseButton"
        icon={renderPlusMinusBtn('minus', isMinimum)}
        type={IconButtonType.NO_BORDER}
        height={15}
        width={35}
        disabled={isMinimum}
        onPress={onDecreaseQty}
      />
      <TouchableOpacity onPress={onInputPress} style={styles.buttonInput}>
        <Text style={styles.calcInput}>{value.toString()}</Text>
      </TouchableOpacity>
      <IconButton
        testID="increaseButton"
        icon={renderPlusMinusBtn('plus', isMaximum)}
        type={IconButtonType.NO_BORDER}
        height={15}
        width={35}
        disabled={isMaximum}
        onPress={onIncreaseQty}
      />
    </View>
  );
};

export default CalcNumericSelector;

CalcNumericSelector.defaultProps = {
  onEndEditing: () => {}
};
