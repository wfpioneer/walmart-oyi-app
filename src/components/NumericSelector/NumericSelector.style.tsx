import { StyleSheet } from 'react-native';
import COLOR from '../../themes/Color';

const styles = StyleSheet.create({
  updateContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    borderRadius: 10,
    borderWidth: 1,
    marginBottom: 4
  },
  updateContainerValid: {
    borderColor: COLOR.GREY_300
  },
  updateContainerInvalid: {
    borderColor: COLOR.RED_900
  },
  input: {
    color: COLOR.MAIN_THEME_COLOR,
    minWidth: '15%',
    textAlign: 'center',
    height: 40
  }
});

export default styles;
