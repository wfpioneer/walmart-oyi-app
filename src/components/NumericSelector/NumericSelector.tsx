import React, { useEffect, useState } from 'react';
import { ActivityIndicator, Text, TextInput, View } from 'react-native';
import MaterialCommunityIcon from 'react-native-vector-icons/MaterialCommunityIcons';
import styles from './NumericSelector.style';
import COLOR from '../../themes/Color';
import Button from '../buttons/Button';
import IconButton from '../buttons/IconButton';
import { numbers, strings } from '../../locales';

interface NumericSelectorProps {
  isValid: boolean;
  onDecreaseQty: any;
  onIncreaseQty: any;
  onTextChange: any;
  value: number;
}

const renderPlusMinusBtn = (name: 'plus' | 'minus') => (
  <MaterialCommunityIcon name={name} color={COLOR.MAIN_THEME_COLOR} size={18} />
);

const NumericSelector = (props: NumericSelectorProps): JSX.Element => {

  return (
    <View
      style={[styles.updateContainer, props.isValid ? styles.updateContainerValid : styles.updateContainerInvalid]}
    >
      <IconButton
        icon={renderPlusMinusBtn('minus')}
        type={IconButton.Type.NO_BORDER}
        height={15}
        width={35}
        onPress={props.onDecreaseQty}
      />
      <TextInput
        style={styles.input}
        keyboardType="numeric"
        onChangeText={props.onTextChange}
      >
        {props.value}
      </TextInput>
      <IconButton
        icon={renderPlusMinusBtn('plus')}
        type={IconButton.Type.NO_BORDER}
        height={15}
        width={35}
        onPress={props.onIncreaseQty}
      />
    </View>
  );
};

export default NumericSelector