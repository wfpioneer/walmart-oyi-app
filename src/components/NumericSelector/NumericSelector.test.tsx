import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import NumericSelector from './NumericSelector';

describe('AddZoneScreen', () => {
  const value = 10;
  const renderer = ShallowRenderer.createRenderer();
  it('NumericSelector with valid input', () => {
    renderer.render(<NumericSelector
      isValid={true}
      onDecreaseQty={jest.fn()}
      onIncreaseQty={jest.fn()}
      onTextChange={jest.fn()}
      value={value}
    />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  it('NumericSelector with invalid input', () => {
    renderer.render(<NumericSelector
      isValid={false}
      onDecreaseQty={jest.fn()}
      onIncreaseQty={jest.fn()}
      onTextChange={jest.fn()}
      value={value}
    />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
