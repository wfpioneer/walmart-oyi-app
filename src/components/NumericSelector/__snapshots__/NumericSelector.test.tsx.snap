// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddZoneScreen NumericSelector minus button is disabled if value is less than or equal to minValue 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 4,
      },
      Object {
        "borderColor": "#e0e0e0",
      },
    ]
  }
>
  <IconButton
    backgroundColor="#0071e9"
    disabled={true}
    height={15}
    icon={
      <mockConstructor
        color="#5FA7F2"
        name="minus"
        size={18}
      />
    }
    onPress={[MockFunction]}
    radius={5}
    type={3}
    width={35}
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "color": "#0071e9",
        "height": 40,
        "minWidth": "15%",
        "textAlign": "center",
      }
    }
    underlineColorAndroid="transparent"
  >
    10
  </TextInput>
  <IconButton
    backgroundColor="#0071e9"
    disabled={false}
    height={15}
    icon={
      <mockConstructor
        color="#0071e9"
        name="plus"
        size={18}
      />
    }
    onPress={[MockFunction]}
    radius={5}
    type={3}
    width={35}
  />
</View>
`;

exports[`AddZoneScreen NumericSelector plus button is disabled if value is greater than or equal to maxValue 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 4,
      },
      Object {
        "borderColor": "#e0e0e0",
      },
    ]
  }
>
  <IconButton
    backgroundColor="#0071e9"
    disabled={false}
    height={15}
    icon={
      <mockConstructor
        color="#0071e9"
        name="minus"
        size={18}
      />
    }
    onPress={[MockFunction]}
    radius={5}
    type={3}
    width={35}
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "color": "#0071e9",
        "height": 40,
        "minWidth": "15%",
        "textAlign": "center",
      }
    }
    underlineColorAndroid="transparent"
  >
    10
  </TextInput>
  <IconButton
    backgroundColor="#0071e9"
    disabled={true}
    height={15}
    icon={
      <mockConstructor
        color="#5FA7F2"
        name="plus"
        size={18}
      />
    }
    onPress={[MockFunction]}
    radius={5}
    type={3}
    width={35}
  />
</View>
`;

exports[`AddZoneScreen NumericSelector with invalid input 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 4,
      },
      Object {
        "borderColor": "#b71c1c",
      },
    ]
  }
>
  <IconButton
    backgroundColor="#0071e9"
    disabled={false}
    height={15}
    icon={
      <mockConstructor
        color="#0071e9"
        name="minus"
        size={18}
      />
    }
    onPress={[MockFunction]}
    radius={5}
    type={3}
    width={35}
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "color": "#0071e9",
        "height": 40,
        "minWidth": "15%",
        "textAlign": "center",
      }
    }
    underlineColorAndroid="transparent"
  >
    10
  </TextInput>
  <IconButton
    backgroundColor="#0071e9"
    disabled={false}
    height={15}
    icon={
      <mockConstructor
        color="#0071e9"
        name="plus"
        size={18}
      />
    }
    onPress={[MockFunction]}
    radius={5}
    type={3}
    width={35}
  />
</View>
`;

exports[`AddZoneScreen NumericSelector with valid input 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 4,
      },
      Object {
        "borderColor": "#e0e0e0",
      },
    ]
  }
>
  <IconButton
    backgroundColor="#0071e9"
    disabled={false}
    height={15}
    icon={
      <mockConstructor
        color="#0071e9"
        name="minus"
        size={18}
      />
    }
    onPress={[MockFunction]}
    radius={5}
    type={3}
    width={35}
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "color": "#0071e9",
        "height": 40,
        "minWidth": "15%",
        "textAlign": "center",
      }
    }
    underlineColorAndroid="transparent"
  >
    10
  </TextInput>
  <IconButton
    backgroundColor="#0071e9"
    disabled={false}
    height={15}
    icon={
      <mockConstructor
        color="#0071e9"
        name="plus"
        size={18}
      />
    }
    onPress={[MockFunction]}
    radius={5}
    type={3}
    width={35}
  />
</View>
`;
